### PR TITLE
Fix documentation in Presto flows

### PR DESCRIPTION
**Pull request by anthonylouisbsb**
# Pull Request Merge Checklist
           
Task backlog identifier related to this PR: ---

---

## As a DEVELOPER, I confirm that this PR **doesn't contain**:
- [x] Code lines with syntax errors
- [x] Methods and variables names that aren't clear or doesn't mean anything
- [x] Code that is out of the standard (Example: **PEP-8** for Python code)
- [x] Files with **imports** errors or without any utility
- [x] **TODOs** left behind
- [x] Tests that are being skipped
- [x] Methods placed on incorrect files or folders (Example: methods that are local, but should be placed on /utils)
- [x] Functions, methods or API responses that are publicly returning sensitive information

## As a DEVELOPER, I confirm that I:
- [x] Followed the Simbiose's standard for branch and commit naming
- [x] Concluded the task and achieved the goals according to the task planning I defined before starting to code
- [x] Added all necessary database changes to the [SQL Migrations](https://github.com/s1mbi0se/sql-migrations)
- [x] Created tests for every new flow and method on this Pull Request
- [x] Added the necessary DEBUG logs
- [x] Checked my code using the Simbiose's Corner Cases Guide
- [x] Merged the **develop** branch to this PR/branch before sending the PR for review
- [x] Solved all possible merge conflicts before sending the PR for review
- [x] Corrected all the problems and code smells identified by the quality tools
- [x] Manually executed the necessary tests locally to make sure my changes didn't break anything
- [x] All unitary and integration tests ran by the CI system are passing
- [x] Check the performance of my changes or additions and they didn't make the system slower (I profiled the code)
- [x] Updated the necessary documentation (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [x] Made the self-review of the code before sending the PR for review
- [x] Sent this PR for a peer-review
- [x] Made all the corrections appointed during the peer-review
- [x] Confirmed that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [x] Checked all GitHub Actions related to this PR that can block my merge

---  

## As the TEAM MANAGER, I confirm that I:
- [ ] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [ ] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [ ] Checked if there is no possible merge conflicts before reviewing the PR
- [ ] Checked if the purpose of the task was reached with this PR
- [ ] Checked that the code was developed following the best practices for the language, with the best possible readability and using the less possible quantity of code lines and complexity. 
- [ ] Checked if all related PRs were created when other components were affected by this task (e.g. database changes on [SQL Migrations](https://github.com/s1mbi0se/sql-migrations))
- [ ] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [ ] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [ ] Checked that code coverage for this PR is higher than 80%.
- [ ] Checked that there are no problems and code smells identified by the quality and security tools
- [ ] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [ ] Checked that all classes, methods and variables on this PR follow the Simbiose’s pattern for this programming language
- [ ] Checked that all unitary and integration tests ran by the CI system are passing
- [ ] Checked that there are no performance issues created by these changes or additions
- [ ] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [ ] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [ ] Checked all GitHub Actions related to this PR that can block the merge

---  

## As the QUALITY RESPONSIBLE, I confirm that I:
- [x] Checked if the Pull Request and commit messages follow the Simbiose’s defined standard 
- [x] Checked if there is no possible merge conflicts before reviewing the PR
- [x] Checked that were created unit, integration and chaos tests for every new flow and method on this Pull Request
- [x] Checked that no sensible information (e.g. users security data, internal security data, etc) was exposed by this PR
- [x] Checked that code coverage for this PR is higher than 80%.
- [x] Checked that there are no problems and code smells identified by the quality and security tools
- [x] Checked that all files were created on the proper folders according to the Simbiose’s defined standard
- [x] Checked that all classes, methods and variables on this PR follow the Simbiose’s code standard for this programming language
- [x] Checked that all unitary and integration tests ran by the CI system are passing
- [ ] Checked that there are no performance issues created by these changes or additions
- [ ] Checked that all documentation is up to date according to this change (Example: OpenAPI, Docstrings, Metadata Database Modeling)
- [ ] Checked that this PR merge will not cause other systems or dependencies to stop working (Example: ShannonDB Python Client and ShannonDB)
- [ ] Checked all GitHub Actions related to this PR that can block the merge

## Important links:
1. [Simbiose Quality Guidelines](https://github.com/s1mbi0se/company/wiki/Quality-Topics#table-of-contents)
2. [Metadata Database Modeling](https://docs.google.com/spreadsheets/d/1Y8_k5xV86kTjgfjbi-PGsJcubUKhWlT2x8ecpnMlyR0/edit?usp=sharing)
3. [Simbiose's Corner Cases Guide](https://www.simbioseventures.com/documentos-e-guias/procedimentos-e-checklists/corner-cases-guide)

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/sql/planner/PartitioningScheme.java [hunk 1]
```diff
@@ -85,6 +85,11 @@ public Partitioning getPartitioning()
         return partitioning;
     }
 
+    /**
+     * Gets the type that the partition is represented.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/sql/planner/PartitioningScheme.java]:
```suggestion
     * Gets the type that the partition is representing.
```

File: presto-main/src/main/java/io/prestosql/sql/planner/PartitioningScheme.java [hunk 2]
```diff
@@ -85,6 +85,11 @@ public Partitioning getPartitioning()
...
+     *
+     * @return the type that the partition is represented
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/sql/planner/PartitioningScheme.java]:
```suggestion
     * @return the type that the partition is representing
```
```suggestion
     * @return the type that the partition is represented
```

File: presto-main/src/main/java/io/prestosql/sql/planner/OrderingScheme.java [hunk 3]
```diff
@@ -50,12 +50,22 @@ public OrderingScheme(@JsonProperty("orderBy") List<Symbol> orderBy, @JsonProper
         this.orderings = ImmutableMap.copyOf(orderings);
     }
 
+    /**
+     * Gets the all ORDER BY clauses.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/sql/planner/OrderingScheme.java:54]:
```suggestion
     * Gets all the defined ORDER BY clauses.
```
Where are these clauses defined?

File: presto-main/src/main/java/io/prestosql/server/protocol/Query.java [hunk 4]
```diff
@@ -319,6 +319,15 @@ public synchronized boolean isClearTransactionId()
         return clearTransactionId;
     }
 
+    /**
+     * Wait for results of te query processing.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/protocol/Query.java]:
```suggestion
     * Waits for results of the query processing.
```

File: presto-main/src/main/java/io/prestosql/server/protocol/Query.java [hunk 5]
```diff
@@ -319,6 +319,15 @@ public synchronized boolean isClearTransactionId()
...
+     *
+     * @param token a token used to create pagination in response
+     * @param wait the maximum time to wait query response
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/protocol/Query.java]:
```suggestion
     * @param wait the maximum time to wait for query response
```

File: presto-main/src/main/java/io/prestosql/server/protocol/Query.java [hunk 6]
```diff
@@ -338,6 +347,11 @@ public synchronized boolean isClearTransactionId()
         return Futures.transform(futureStateChange, ignored -> getNextResult(token, uriInfo, targetResultSize), resultsProcessorExecutor);
     }
 
+    /**
+     * Gets the result of the query processing.
+     *
+     * @return the result of query processing
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/protocol/Query.java]:
```suggestion
     * @return the result of the query processing
```

File: presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java [hunk 7]
```diff
@@ -143,6 +143,17 @@ public void stop()
         queryPurger.shutdownNow();
     }
 
+    /**
+     * Gets the results for a executing query.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java]:
```suggestion
     * Gets the results for an executing query.
```

File: presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java [hunk 8]
```diff
@@ -143,6 +143,17 @@ public void stop()
...
+     *
+     * @param queryId the query's identifier
+     * @param slug a identifier for query in URI
+     * @param token a token used to create pagination in response
+     * @param maxWait the maximum time to wait query response
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java]:
```suggestion
     * @param maxWait the maximum time to wait for query response
```

File: presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java [hunk 9]
```diff
@@ -198,6 +209,16 @@ protected Query getQuery(QueryId queryId, String slug, long token)
         return query;
     }
 
+    /**
+     * Retrieves the query result asynchronously.
+     *
+     * @param query the query to be processed
+     * @param token a token used to create pagination in response
+     * @param maxWait the maximum time to wait query response
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/protocol/ExecutingStatementResource.java]:
```suggestion
     * @param maxWait the maximum time to wait for query response
```

File: presto-main/src/main/java/io/prestosql/server/TaskResource.java [hunk 10]
```diff
@@ -252,6 +252,13 @@ public TaskInfo deleteTask(
         return taskInfo;
     }
 
+    /**
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/TaskResource.java:255]: No summary?

File: presto-main/src/main/java/io/prestosql/server/TaskResource.java [hunk 11]
```diff
@@ -357,6 +364,12 @@ private static boolean shouldSummarize(UriInfo uriInfo)
         return uriInfo.getQueryParameters().containsKey("summarize");
     }
 
+    /**
+     * Gets a random time between the max wait time and half of the value.
+     *
+     * @param waitTime the max time to wait query to be processed
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/server/TaskResource.java]:
```suggestion
     * @param waitTime the max time to wait for the query to be processed
```

File: presto-main/src/main/java/io/prestosql/metadata/HandleResolver.java [hunk 12]
```diff
@@ -249,6 +249,14 @@ public MaterializedHandleResolver(ConnectorHandleResolver resolver)
             return insertTableHandle;
         }
 
+        /**
+         * Gets the class used to tell the partition scheme of the query execution.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/metadata/HandleResolver.java]:
```suggestion
         * Gets the class used to inform the partition scheme about the query execution.
```

File: presto-main/src/main/java/io/prestosql/metadata/HandleResolver.java [hunk 13]
```diff
@@ -249,6 +249,14 @@ public MaterializedHandleResolver(ConnectorHandleResolver resolver)
...
+         * <p>
+         * The partition scheme shows if the query must be executed by one node or must be
+         * distributed through the instances.
+         *
+         * @return the class used to tell the partition scheme of the query execution
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/metadata/HandleResolver.java]:
```suggestion
         * @return the class used to inform the partition scheme about the query execution
```
```suggestion
         * @return the class used to tell the partition scheme of the query execution
```

File: presto-main/src/main/java/io/prestosql/metadata/DynamicCatalogStoreRoundRobin.java [hunk 14]
```diff
@@ -43,6 +50,16 @@ public String getServer()
         return target;
     }
 
+    /**
+     * Gets the quantity of running API instances.
+     * <p>
+     * Presto executes requests periodically to API to retrieve information about
+     * data connections(catalogs). If the instance that server executed the request is down,
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/metadata/DynamicCatalogStoreRoundRobin.java]:
```suggestion
     * data connections (catalogs). If the instance that server executed the request is down,
```

File: presto-main/src/main/java/io/prestosql/metadata/DynamicCatalogStoreRoundRobin.java [hunk 15]
```diff
@@ -43,6 +50,16 @@ public String getServer()
...
+     * the server must execute the request to the next instance until the request is completed
+     * or the server tried to execute a request to all API instances.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/metadata/DynamicCatalogStoreRoundRobin.java]:
```suggestion
     * or if the server already tried to execute a request to all API instances.
```

File: presto-main/src/main/java/io/prestosql/metadata/CatalogManager.java [hunk 16]
```diff
@@ -44,6 +44,11 @@ public synchronized void registerCatalog(Catalog catalog)
                 .map(Catalog::getConnectorCatalogName);
     }
 
+    /**
+     * Gets the list of all existed catalogs inside the server.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/metadata/CatalogManager.java]:
```suggestion
     * Gets the list of all existing catalogs inside the server.
```

File: presto-main/src/main/java/io/prestosql/metadata/CatalogManager.java [hunk 17]
```diff
@@ -44,6 +44,11 @@ public synchronized void registerCatalog(Catalog catalog)
...
+     *
+     * @return the list of all existed catalogs inside the server
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/metadata/CatalogManager.java]:
```suggestion
     * @return the list of all existing catalogs inside the server
```

File: presto-main/src/main/java/io/prestosql/memory/MemoryPoolAssignmentsRequest.java [hunk 18]
```diff
@@ -36,6 +36,14 @@ public MemoryPoolAssignmentsRequest(@JsonProperty("coordinatorId") String coordi
         this.assignments = ImmutableList.copyOf(requireNonNull(assignments, "assignments is null"));
     }
 
+    /**
+     * Gets the coordinator identifier.
+     * <p>
+     * It is used to discover which machine(node) is the
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/memory/MemoryPoolAssignmentsRequest.java]:
```suggestion
     * Used to discover which machine (node) is the
```

File: presto-main/src/main/java/io/prestosql/dispatcher/LocalDispatchQuery.java [hunk 19]
```diff
@@ -156,6 +159,11 @@ public DateTime getLastHeartbeat()
         return stateMachine.getLastHeartbeat();
     }
 
+    /**
+     * Gets the futures that represent the query dispatch operation to be executed
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/dispatcher/LocalDispatchQuery.java]:
```suggestion
     * Gets the futures that represent the query dispatch operation to be executed.
```

File: presto-main/src/main/java/io/prestosql/execution/SqlTask.java [hunk 20]
```diff
@@ -195,16 +208,34 @@ public SqlTaskIoStats getIoStats()
...
     public String getTaskInstanceId()
     {
         return taskInstanceId;
     }
 
+    /**
+     * Records the last time that Presto act on the query.
+     * <p>
+     * As an example: every time that a request is executed to know the query status in Presto,
+     * the last heartbeat is increases.
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/execution/SqlTask.java]:
```suggestion
     * the last heartbeat increases.
```

File: presto-main/src/main/java/io/prestosql/execution/SqlTask.java [hunk 21]
```diff
@@ -229,6 +265,16 @@ public TaskStatus getTaskStatus()
         }
     }
 
+    /**
+     * Gets an object with task metadata.
+     * <p>
+     * It will checks if the holder contains any executing task and will
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/execution/SqlTask.java]:
```suggestion
     * Checks if the holder contains any executing task and will
```

File: presto-main/src/main/java/io/prestosql/execution/StateMachine.java [hunk 22]
```diff
@@ -316,6 +322,14 @@ public String toString()
         return get().toString();
     }
 
+    /**
+     * Executes a generic thread.
+     * <p>
+     * It executes the thread and if the server is shutting down
```
Comment (jpedroantunes) [inline presto-main/src/main/java/io/prestosql/execution/StateMachine.java]:
```suggestion
     * Executes the thread and if the server is shutting down
```

**Review by jpedroantunes — CHANGES_REQUESTED** [review]

File: presto-main/src/main/java/io/prestosql/sql/planner/OrderingScheme.java (hunk 3 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/sql/planner/OrderingScheme.java:54]: Description added

File: presto-main/src/main/java/io/prestosql/server/TaskResource.java (hunk 10 above)
Comment (anthonylouisbsb) [reply presto-main/src/main/java/io/prestosql/server/TaskResource.java:255]: added
